src/App.js

```diff
@@ -1,85 +1,217 @@
+
+    newTodo.value = newTodoValue;
+    newTodo.id = currentId;
+    newTodo.completed = false;
+    this.setState(state => ({
+      todos: [...state.todos, newTodo],
+      currentId: state.currentId + 1,
+      selectAll: false,
+    }));
+  }
+
+  choosePage = (list) => {
+    const { todos } = this.state;
+
+    switch (list) {
+      case 'all':
+        return todos;
+
+      case 'active':
+        return todos.filter(todo => !todo.completed);
+
+      case 'completed':
+        return todos.filter(todo => todo.completed);
+
+      default:
+        return todos;
+    }
+  }
+
+  changeStatusComplete = (todoId) => {
+    this.setState(state => ({
+      todos: state.todos.map(todo => ({
+        ...todo,
+        completed: todo.id === todoId
+          ? !todo.completed
+          : todo.completed,
+      })),
+    }), () => this.filterTodos());
+  }
+
+  destroyTodo = (todoId) => {
+    const { todos } = this.state;
+    const destroyedTodo = todos.filter(todo => todo.id !== todoId);
+
+    localStorage.removeItem(`${todoId}`);
+
+    this.setState(() => ({ todos: destroyedTodo }));
+
+    setTimeout(() => {
+      localStorage.clear();
+      this.state.todos.forEach((todo, i) => (
+        localStorage.setItem(`${i + 1}`, JSON.stringify({ ...todo }))
+      ));
+      if (!this.state.todos.length) {
+        this.setState(({ currentId: 1 }));
+      } else {
+        const lastItem = this.state.todos.length - 1;
+        const newCurrentId = this.state.todos[lastItem].id + 1;
+
+        this.setState(({ currentId: newCurrentId }));
+      }
+
+      this.filterTodos();
+    }, 0);
+  }
+
+  changePage = (changeTo) => {
+    this.setState(({ activeList: changeTo }));
+  }
+
+  filterTodos = () => {
+    const { todos } = this.state;
+    const completedTodo = todos.filter(todo => todo.completed);
+
+    todos.length === completedTodo.length
+      ? this.setState(() => ({ selectAll: true }))
+      : this.setState(() => ({ selectAll: false }));
+  }
+
+  clearCompletedTodo = () => {
+    const { todos } = this.state;
+    const clearCompleted = todos.filter(todo => !todo.completed);
+
+    this.setState(() => ({
+      todos: clearCompleted,
+    }));
+  }
+
+  selectAllTodo = () => {
+    const { selectAll } = this.state;
+
+    this.setState(state => ({
+      todos: [...state.todos.map(todo => ({
+        ...todo,
+        completed: !selectAll,
+      }))],
+      selectAll: !state.selectAll,
+    }), () => this.filterTodos());
+  }
+
+  changeTodoValue = (todoId, newValue) => {
+    const { todos } = this.state;
+
+    this.setState(() => ({
+      todos: todos.map(todo => ({
+        ...todo,
+        value: todo.id === todoId
+          ? newValue
+          : todo.value,
+      })),
+    }), () => {
+      localStorage.clear();
+      this.state.todos.forEach((todo, i) => (
+        localStorage.setItem(`${i + 1}`, JSON.stringify({ ...todo }))
+      ));
+    });
+  }
+
+  valueOfLocalStorage = () => {
+    const user = [];
```

Comment:
переменная user хранит в себе массив todos?
```suggestion
    const usersTodos = [];
```
или 
```suggestion
    const localTodos = [];
```

Reply:
С радостью